about.php

```diff
@@ -2,6 +2,7 @@
 <html>
 <head>
 	<title>OpenFlights: About</title>
+    <meta name="version" content="<?php echo(`git rev-parse HEAD`); ?>">
```

Comment:
Important note: `git` will refuse to run if the repo is not owned by $USER.

Reply:
Can we indent it consistently please? ;)

Reply:
Taaaaaabs! \*shakes fist\*

More importantly, I'm afraid this won't work with the setup described in [INSTALL](https://github.com/jpatokal/openflights/blob/master/INSTALL). The instructions do the checkout as `openflights` but the server will be running as `www-data` so `git` will refuse to run. In my local testing the files were owned by `www-data`. Womp Womp.

Here are some options:

1. We could silence the ownership warning by running this as part of the install:
   ```sh
   sudo -u www-data git config --global --add safe.directory /var/www/openflights
   ```
   This would also have to be run on production. I think this makes it less likely to happen. :)

2. A post-checkout hook dumping the hash to a file would also work but, again, this also requires enabling the hook on the clone.

3. My previous attempt (`echo(file_get_contents(".git/refs/heads/master")`) doesn't need this but assumes `master` is the checked out branch.

4. A more general version would run `.git/$(cat .git/HEAD | cut -d' ' -f2)` which now works with different branches (but not a detached HEAD).

@reedy, thoughts or preferences? Do you know of a better way?

Reply:
We've got a class in MediaWiki for dealing with these various edge cases, unfortunately it's not very portable - https://github.com/wikimedia/mediawiki/blob/master/includes/GitInfo.php

I suspect the answer here may be "don't just try and do it as a one liner".

But also, most people aren't probably going to be running their own version of openflights (unlikes someone might with MediaWiki), so potentially something that works trivially enough on a dev install, but will also work on the actual deployment (I don't have access, so I don't really know how it looks).